src/run.js

```diff
@@ -9,6 +9,15 @@ const cheerio = require('cheerio');
 const utils = require('./utils');
 const url = require('url');
 
+const isOk = status => status === 200 || status === 304;
```

Comment:
We should probably stick closer to how [puppeteer does it](https://github.com/GoogleChrome/puppeteer/blob/38c6873fbbc469fe422977131bf2da8e75ec9228/lib/NetworkManager.js#L554)

Perhaps, instead of our own `isOk` function, we do something like:
```javascript
if (!(response.ok() || response.status() === 304)) {
```

Kinda strange the puppeteer doesn't consider 304 to be "OK". 

Also, we could do something like 
```javascript
if (response.status() >= 300 && response.headers().location) {
   ... it wants to redirect...
```

Reply:
note: this is how ok function looks like

```js
  ok() {
    return this._status === 0 || (this._status >= 200 && this._status <= 299);
  }
```